pkg/proxy/handlers/spacelister_get_test.go

```diff
@@ -711,20 +706,33 @@ func TestGetUserWorkspace(t *testing.T) {
 	for k, tc := range tests {
 		t.Run(k, func(t *testing.T) {
 			// given
+			fakeClient := test.NewFakeClient(t,
+				// space
+				fake.NewSpace("batman", "member-1", "batman"),
+				fake.NewSpace("robin", "member-1", "robin"),
+
+				fake.NewSpaceBinding("robin-1", "robin", "robin", "admin"),
+				// 2 spacebindings to force the error
+				fake.NewSpaceBinding("batman-1", "batman", "batman", "admin"),
+				fake.NewSpaceBinding("batman-2", "batman", "batman", "maintainer"),
+				fake.NewSpaceBinding("community-robin", toolchainv1alpha1.KubesawAuthenticatedUsername, "robin", "viewer"),
+			)
```

Comment:
not sure if there was any specific reason for declaring this `fakeClient` for each test iteration. Maybe we could move it back outside ( not a big performance improvement but still ) .

Reply:
the reason for moving it there was the mock functions - some of the iterations mock one of the FakeClient functions and we want to do it in the context of the iteration only, so it doesn't affect other test cases.

Reply:
ok makes sense, thanks for pointing that out.